docs/synapse/deploymentguide.rst

```diff
@@ -0,0 +1,355 @@
+.. _deploymentguide:
+
+Synapse Deployment Guide
+########################
+
+Introduction
+============
+
+This step-by-step guide will walk you through a production-ready Synapse deployment. Services will be
+configured to register with ``AHA`` for service discovery and to prepare for future devops tasks such
+as promoting a mirror to leader and provisioning future Synapse Advanced Power-Ups.
+
+This guide will also walk you through deploying all Synapse services using TLS to authenticate both
+servers and clients using client-certificates to minimize the need for secrets management by eliminating
+passwords from all telepath URLs.
+
+For the purposes of this guide, we will use ``docker-compose`` as a light-weight orchestration mechanism.
+The steps, configurations, and volume mapping guidance given in this guide apply equally to other container
+orchestration mechanisms such as Kubernetes but for simplicity's sake, this guide will only cover
+``docker-compose`` based deployments.
+
+Prepare your Hosts
+==================
+
+Ensure that you have an updated install of docker_ and docker-compose_.
+
+In order to help you run the Synapse service containers as a non-root user, Synapse service docker containers
+have been preconfigured with a user named ``synuser`` with UID ``999``. You may replace ``999`` in the configs
+below, but keep in mind that doing so will result in the container not having a name for the user. We recommend
+that you do **not** use the Linux user ``nobody`` for this purpose.
+
+Default kernel parameters on most Linux distributions are not optimized for database performance. We recommend
+adding the following lines to ``/etc/sysctl.conf`` on all systems being used to host Synapse services::
+
+    vm.swappiness=10
```

Comment:
These settings (plus a couple more) are in devops_general.rst. Maybe one should be a link to the other so they are always in sync?

Reply:
The devops_general.rst is going away, but i will make it link to the new place that lives